### PR TITLE
Fix issue #1063

### LINKED ISSUE #1063: Feature Request: Edit images using quadrilateral selection in image to PDF
Describe your issue here.
When I add images and choose `Edit Images`, the image can only crop by a rectangle. If a quadrilateral can be used to crop the image, it will be much more convenient to process documents and other images.
image

### PR DESCRIPTION
**Pull request by VinAjuu**
modified:   app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java
	new file:   app/src/main/res/layout/activity_crop_by_quadrilater.xml
	modified:   app/src/main/res/layout/activity_crop_image_activity.xml

# Description
Using a lib to fix the issue. Create a new view and need a new dependency.
Dependency:  com.github.pqpo:SmartCropper:v2.1.3
Add a new button point to the new view.
image
By clicking the `Crop` button, the new view is like.
image
The pdf is like:
image

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
If there are any UI change, please include the screenshots also.

Fixes #1063 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Tested on `Pixel 2 API 30` in Android Studio
Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by VinAjuu** [comment]
Fix the issue of Complexity by refactoring.

File: app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java [hunk 1]
```diff
@@ -168,6 +217,7 @@ private void setUpCropImageView() {
 
     /**
      * Set image in crop image view & increment counters
+     *
```
Comment (codegsaini) [inline app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:220]: Revert  these unnecessary changes!

File: app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java [hunk 2]
```diff
@@ -112,7 +161,7 @@ public void nextImageClicked() {
     }
 
     public void prevImgBtnClicked() {
-        if ( mImages.size() == 0)
+        if (mImages.size() == 0)
```
Comment (codegsaini) [inline app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:164]: Revert unnecessary...

File: app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java [hunk 3]
```diff
@@ -94,13 +118,38 @@ public void cropButtonClicked() {
...
+    }
+
     public void rotateButtonClicked() {
         mCurrentImageEdited = true;
         mCropImageView.rotateImage(90);
     }
 
     public void nextImageClicked() {
-        if ( mImages.size() == 0)
+        if (mImages.size() == 0)
```
Comment (codegsaini) [inline app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:152]: I know these unnecessary changes are not any big deal for now, but they can confuse if anyone watch these commit later. These all beautification process and other thing deserve separate commit.

**Review by codegsaini — APPROVED** [review]
~~Ok, nice work :) @Swati4star can consider reviewing it before merging.
Keep contributing!~~ I will test it before approving...

**Comment by VinAjuu** [comment]
Unnecessary changes have been reverted. Sorry for the inconvenience caused by them. They are created by the refactoring of IDE.

File: app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java [hunk 4]
```diff
@@ -133,6 +133,7 @@ public void cropOKButtonClicked() {
         Bitmap crop = ivCrop.crop();
         mCropImageView.setImageBitmap(crop);
         setContentView(mMainView);
+        cropButtonClicked();
```
Comment (codegsaini) [inline app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:136]: `cropButtonClicked()` method inside `cropOKButtonClicked()` is little confusing!!! Can you explain please...

Comment (VinAjuu) [reply app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:136]: Use the cropButtonClicked() to store the image with the URI. Because after you crop with a quadrilateral selection, the image should be saved immediately. If not saved, the URI of the image will be null and an exception will occur.

Comment (VinAjuu) [reply app/src/main/java/swati4star/createpdf/activity/CropImageActivity.java:136]: cropButtonClicked() is actually a method to save the cropped image as a file.

**Comment by VinAjuu** [comment]
@codegsaini Thanks for testing the code. If you have any questions, just let me know.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -580,6 +580,7 @@
     <string name="storage_manager_permission_rationale">In Android 11, you need to allow a special storage permission</string>
     <string name="close_app_text">Close App</string>
     <string name="allow_text">Allow</string>
+    <string name="string_mode">\@string/mode</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml:583]: what is this one being used for?

**Review by Swati4star — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (VinAjuu) [reply app/src/main/res/values/strings.xml:583]: This is generated by IDE automatically. I will modify the file later.

Comment (VinAjuu) [reply app/src/main/res/values/strings.xml:583]: Actually, this string is not used in code.

**Comment by VinAjuu** [comment]
I modified the button text in the format of `@string/content`, and add the content string into string.xml,

**Comment by VinAjuu** [comment]
@Swati4star There is a `changes request` label, could you please give more detail about the changes requested.
